src/lsm/compaction.zig

```diff
@@ -41,6 +41,7 @@ const constants = @import("../constants.zig");
 
 const stdx = @import("../stdx.zig");
 const maybe = stdx.maybe;
+const vsr = @import("../vsr.zig");
```

Comment:
Yeah, I 100% agree that it would be nice to keep LSM unaware of VSR's checkpoint schedule.

> I wonder if it would be worth retaining this design by pulling the decision about checkpoint up, and pass flush_memory_table: bool parameter to the LSM? It feels like we might need something like that anyway once we get to snapshots?

I assume you mean as a parameter for `compact()`? Since `.compact()` is called for every beat, not just every bar/half-bar, it might make more sense to have a separate `.checkpoint_soon()` function which is called exactly one bar before `.checkpoint()`.

...That said, I am going to punt this to a TODO b/c it is a weekend/holiday.